java/vector/src/main/java/org/apache/arrow/vector/BaseDataValueVector.java

```diff
@@ -30,6 +30,9 @@
 
   protected final static byte[] emptyByteArray = new byte[]{}; // Nullable vectors use this
 
+  /** maximum extra size at the end of the buffer */
+  private static final int MAX_BUFFER_PADDING = 0; // TODO: check
```

Comment:
64, if I'm reading your code right. from https://github.com/apache/arrow/blob/master/format/Layout.md#requirements-goals-and-non-goals we have "All contiguous memory buffers are aligned at 64-byte boundaries and padded to a multiple of 64 bytes". 

As an aside, I think we can relax the alignment requirement because SIMD will still work on data that is only word-aligned, but with a slight performance degradation. We need the padding though